### DISCUSSION TITLE
How to implement data exchange  between mini-programs?

### DISCUSSION THREAD

**Discussion by spyx1** [zepp-health/zeppos-forum#297] [Q&A]
Hello. I can’t find in the documentation how to implement data exchange (json) between mini-programs. localstorage and files (/assets, /data) are isolated by one mini-program.
Can I access a file from another app? Is it possible to get the full path to a file so that it can be read from any mini-program?
Or some other modes of existence `could` be created between the mini-programs and the dial.

**Comment by spyx1** [comment]
Zepp os 2

**Comment by MyoungXUE** [comment]
Zepp OS 3 implemented this function, @zos/fs module `openSync` can specify the Mini Program ID, access to the target Mini Program /data directory files.

https://docs.zepp.com/docs/reference/device-app-api/newAPI/fs/openSync/

What is your device? GTR4 has been gradually updated to Zepp OS 3.

**Reply by jdemaeyer** [reply]
@MyoungXUE I am trying to implement two-way communication. Reading from another app's data directory with the `appId` option as you suggest works great, but it doesn't seem to work for writing into another app's data directory. Is there an undocumented restriction that the `appId` option only works for reading?

(I'm using API 4.0 on an Active 2 NFC)

**Comment by spyx1** [comment]
My device bip 5.

**Reply by JulianBessone** [reply]
Dear @spyx1,

You can try with this... 

https://github.com/silver-zepp/zeppos-easy-storage
